glusterd2 panics on start up

When starting glusterd2 it panics trying to load/check xlators.
I'm probably triggering this by having an "incorrect" environment but I think this was worth a report because a panic is a fairly unfriendly kind of error.


```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xc pc=0x449ff0]

goroutine 1 [running]:
github.com/gluster/glusterd2/glusterd2/xlator._Cfunc_GoString(0xc, 0x2, 0x2)
        github.com/gluster/glusterd2/glusterd2/xlator/_obj/_cgo_gotypes.go:99 +0x2b
github.com/gluster/glusterd2/glusterd2/xlator.structifyOption(0xc420724000, 0x2)
        /var/tmp/gtg-wrapper/src/github.com/gluster/glusterd2/glusterd2/xlator/load.go:62 +0xbe8
github.com/gluster/glusterd2/glusterd2/xlator.loadXlator(0xc420099050, 0x28, 0x0, 0x0, 0x0)
        /var/tmp/gtg-wrapper/src/github.com/gluster/glusterd2/glusterd2/xlator/load.go:137 +0x64c
github.com/gluster/glusterd2/glusterd2/xlator.loadAllXlators.func1(0xc420099050, 0x28, 0x17ce1e0, 0xc42021d5f0, 0x0, 0x0, 0x0, 0x0)
        /var/tmp/gtg-wrapper/src/github.com/gluster/glusterd2/glusterd2/xlator/load.go:195 +0x9e
path/filepath.walk(0xc420099050, 0x28, 0x17ce1e0, 0xc42021d5f0, 0xc42069da60, 0x0, 0x0)
        /usr/lib/golang/src/path/filepath/path.go:356 +0x81
path/filepath.walk(0xc4207143c0, 0x20, 0x17ce1e0, 0xc42021d520, 0xc42069da60, 0x0, 0x0)
        /usr/lib/golang/src/path/filepath/path.go:381 +0x3a0
path/filepath.walk(0xc420098f90, 0x1b, 0x17ce1e0, 0xc42021d450, 0xc42069da60, 0x0, 0x10)
        /usr/lib/golang/src/path/filepath/path.go:381 +0x3a0
path/filepath.Walk(0xc420098f90, 0x1b, 0xc42069da60, 0x0, 0xc42069b740)
        /usr/lib/golang/src/path/filepath/path.go:403 +0x11d
github.com/gluster/glusterd2/glusterd2/xlator.loadAllXlators(0x7d9b75, 0xc42069ea60, 0x0)
        /var/tmp/gtg-wrapper/src/github.com/gluster/glusterd2/glusterd2/xlator/load.go:204 +0x143
github.com/gluster/glusterd2/glusterd2/xlator.Load(0x0, 0x0)
        /var/tmp/gtg-wrapper/src/github.com/gluster/glusterd2/glusterd2/xlator/global.go:18 +0x26
main.main()
        /var/tmp/gtg-wrapper/src/github.com/gluster/glusterd2/glusterd2/main.go:95 +0x540
```

I added a Println of the object triggering the panic:
```
&{[0x7ffbd05f689a <nil> <nil> <nil>] 12 [0 0 0 0] 0 0 [<nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil>] <nil> <nil> 0 [0 3495913644 32763 0] [0 0 0 0] 0 [0xc <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil> <nil>] <nil> 0 [0 0 0 0]}
```

My thought would be to add some sort of error checking or validation to the structifyOption func and have it return an error in case that the struct is not compatible with glusterd2. I tried to find some more information about the struct itself but wasn't able to get much. But I had thought that checking for valid values in min, max, description and/or default_value before unpacking the tags or keys might help.

Please let me know if I can provide any other information.

--- a/glusterd2/xlator/global.go
+++ b/glusterd2/xlator/global.go
@@ -1,7 +1,12 @@
 package xlator
 
 import (
+	"fmt"
+	"runtime/debug"
+
 	"github.com/gluster/glusterd2/glusterd2/xlator/options"
+
+	log "github.com/sirupsen/logrus"
 )
 
 var (
@@ -15,6 +20,15 @@ var (
 // Load load all available xlators and intializes the xlators and options maps
 func Load() (err error) {
 
+	defer func() {
+		if r := recover(); r != nil {
+			log.Info(string(debug.Stack()))
+			err = fmt.Errorf("recover()ed at xlator.Load(): %s", r)
+			log.Error("Your version of glusterfs is incomaptible. ",
+				"Please install latest glusterfs from source (branch: master)")
+		}
+	}()
+
 	xls, err := loadAllXlators()
 	if err != nil {
 		return
@@ -23,8 +37,9 @@ func Load() (err error) {
 
 	loadOptions()
 
-	if err := registerAllValidations(); err != nil {
-		return err
+	err = registerAllValidations()
+	if err != nil {
+		return
 	}
 
 	return

